lib/ex_wire/framing/frame.ex

```diff
@@ -0,0 +1,183 @@
+defmodule ExWire.Framing.Frame do
+  @moduledoc """
+  Handles framing a message for transport in RLPx.
+
+  This is defined in the [RLPx docs[(https://github.com/ethereum/devp2p/blob/master/rlpx.md)
+  under Framing section.
+
+  TODO: Handle multi-frame packets, etc.
+  TODO: Add tests, etc.
+  """
+
+  alias ExthCrypto.MAC
+  alias ExthCrypto.AES
+  alias ExWire.Framing.Secrets
+
+  @type frame :: binary()
+
+  @spec frame(integer(), ExRLP.t, Secrets.t) :: {frame, Secrets.t}
+  def frame(packet_type, packet_data, frame_secrets=%Secrets{egress_mac: egress_mac, encoder_stream: encoder_stream, mac_encoder: mac_encoder, mac_secret: mac_secret}) do
```

Comment:
Is there any way to break this up a little more?

Maybe pull some of the the padding stuff out into a helper?

Reply:
I can probably do so-- this code was supposed to be based off the spec comments, but it ended up getting built off of translating code from Geth / Parity, hence why it doesn't feel very idiomatic. I can probably fix that a bunch.